ci/do_ci.sh

```diff
@@ -186,6 +186,7 @@ elif [[ "$1" == "bazel.compile_time_options" ]]; then
     --define boringssl=fips \
     --define log_debug_assert_in_release=enabled \
     --define tcmalloc=debug \
+    --define quiche=enabled \
```

Comment:
What does this define do? Doesn't this just get enabled automatically when the QUIC listener extension is compiled?

Reply:
Currently this controls whether QUIC listener-related tests are built and run. The short history is that QUICHE platform code requires a Bazel bugfix flag (--experimental_remap_main_repo) to be set in order to successfully link with Envoy libraries, so @htuch [suggested](https://github.com/envoyproxy/envoy/pull/5767#issuecomment-459141802) that we avoid building QUICHE by default (at least, until the flag becomes default behavior), to avoid poor developer experience.  Since then, the Bazel flag has been enabled in the default .bazelrc (#5767), so it should be ok to enable compilation/testing of QUICHE here, though we probably still do not want to set quiche=enabled for developer local builds.

Tagging @wu-bin and @danzh2010 , just as a heads up.

Reply:
OK makes sense. Can we add a TODO comment here that we eventually want to remove this from compile_time_options and have this be part of the standard build?

Reply:
`TODO` added (for you, @mpwarres, hope that's OK).

Reply:
> `TODO` added (for you, @mpwarres, hope that's OK).

Yup, SGTM